.github/workflows/build_wheels.yml

```diff
@@ -0,0 +1,54 @@
+name: Build and upload to PyPI
+
+on: [push, pull_request]
```

Comment:
do we really need to do this on every push? It's about 30 minutes of building wheels and we use the resources for free.
I'm leaning towards building for every pull request and release.
Did you know that github action can be triggered on [release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release)?

Reply:
Oh I didn't know that about releases - will add.

Agree, we could be more conscientious of the generous free resources github actions is allowing us to use - will change to run on PR and release.